state/execution.go

```diff
@@ -76,12 +76,28 @@ func execBlockOnProxyApp(txEventPublisher types.TxEventPublisher, proxyAppConn p
 	}
 	proxyAppConn.SetResponseCallback(proxyCb)
 
+	// determine validators who did not sign last block
+	absentVals := make([]int32, 0)
+	for valA, _ := range lastValidators.Validators {
```

Comment:
Aren't the precommits in order by validator?
I vaguely remember something in the light client, that if slot X was nil, then validator X didn't sign.

This looks correct, but I think it can be made more efficient (O(N*N) here, even if N is only 100, still...)

Reply:
Good question

Reply:
That's right. The `votes` array in a `VoteSet` and the `Precommits` in a `Commit` should always be as large as the validtor set, possibly containing some nils. So we can just look for the nil ones and send that index